docs/core/extensions/globalization-icu.md

```diff
@@ -305,3 +323,34 @@ The following APIs are supported with limitations:
+
+### Windows
+
+On Windows, .NET follow the following steps to initialize globalization:
+- Check whether [Globalization Invariant Mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) is enabled. When this mode is active, .NET bypasses loading the ICU library and avoids using NLS APIs. Instead, it relies on built-in invariant culture data, ensuring that behavior remains fully independent of the operating system and the ICU library.
+
+- Check whether [NLS mode](#use-nls-instead-of-icu) is enabled. If enabled, .NET will skip loading the ICU library and instead rely on Windows [NLS](https://learn.microsoft.com/en-us/windows/win32/intl/national-language-support) APIs for globalization support.
+
+- Check whether the [app-local ICU](#app-local-icu) feature is enabled. If it is, .NET will attempt to load the ICU library from the application directory by appending the specified version to the library names. For instance, if the version is 72.1, .NET will first try to load `icuuc72.dll`, `icuin72.dll`, and `icudt72.dll`. If these libraries cannot be loaded, it will then attempt to load `icuuc72.1.dll`, `icuin72.1.dll`, and `icudt72.1.dll`. If none of the libraries are found, the process will terminate with an error message such as: `Failed to load app-local ICU: {library name}`.
+
+- If none of the preceding conditions are satisfied, .NET will attempt to load the ICU library from the system directory. It first tries to load `icu.dll`. If this library is unavailable, it will then attempt to load `icuuc.dll` and `icuin.dll`from the system directory. If any of these libraries are not found, the runtime will fall back to using NLS APIs for globalization support.
+
+> [!NOTE]
+> NLS APIs are always available in all Windows versions, so .NET can always rely falling back on them for globalization support.
+
+### Linux
+
+- Check whether [Globalization Invariant Mode](https://github.com/dotnet/runtime/blob/main/docs/design/features/globalization-invariant-mode.md) is enabled. When this mode is active, .NET bypasses loading the ICU library. Instead, it relies on built-in invariant culture data, ensuring that behavior remains fully independent of the operating system and the ICU library.
+- Check whether the [app-local ICU](#app-local-icu) feature is enabled. If it is, .NET will attempt to load the ICU library from the application directory by appending the specified version to the library names. For instance, if the version is 68.2.0.9, .NET will try to load `libicuuc.so.68.2.0.9` and `libicui18n.so.68.2.0.9`. If any of the libraries are found, the process will terminate with an error message such as: `Failed to load app-local ICU: {library name}`.
+- Check if the `DOTNET_ICU_VERSION_OVERRIDE` environment variable is set. If it is, .NET will attempt to load the specified version of ICU as desctibed in [Loading Specific ICU Version on Linux](#loading-specific-icu-version-on-linux).
```

Comment:
This step applies for Microsoft "portable" builds only. For example, it won't work if you install .NET on Ubuntu from Ubuntu package feed.

Reply:
Do we have some doc defining what portable and non-portable builds are? I am trying to know if I mention portable build will be obvious to the readers to understand.

Reply:
In user facing docs, I think we can say "Microsoft provided build" and "Linux distro provided build".

(Portable vs. non-portable is a concept of dotnet/runtime build script https://github.com/dotnet/runtime/blob/60a29248bbb501dd037d2984daea3635cf20b412/eng/build.sh#L83 but that is not something to explain in user facing docs like this one.)

Reply:
I have added a note under the section `Load specific ICU version on Linux` which describing `DOTNET_ICU_VERSION_OVERRIDE`.